identity/aziot-identityd/src/identity.rs

```diff
@@ -736,15 +751,33 @@ impl IdentityManager {
         &self,
         identity_pk: &str,
         identity_cert: &str,
-        subject: &str,
-    ) -> Result<(), Error> {
+        subject: Option<&str>,
+    ) -> Result<Option<Result<String, Error>>, Error> {
         // Retrieve existing cert and check it for expiry.
-        let device_id_cert = match self.cert_client.get_cert(identity_cert).await {
+        let (device_id_cert, old_cert_subject_name) = match self
+            .cert_client
+            .get_cert(identity_cert)
+            .await
+        {
             Ok(pem) => {
                 let cert = openssl::x509::X509::from_pem(&pem).map_err(|err| {
                     Error::Internal(InternalError::CreateCertificate(Box::new(err)))
                 })?;
                 let cert_expiration = cert.as_ref().not_after();
+                let cert_subject = cert.as_ref().subject_name();
+                let cert_subject = match cert_subject.entries_by_nid(openssl::nid::Nid::COMMONNAME).next() {
+                    Some(common_name_entry) => {
+                        match String::from_utf8(common_name_entry.data().as_slice().into()) {
```

Comment:
This can be expressed more concisely with `map_err`.

Reply:
Fixed the match at cert_subject level